Dockerfile

```diff
@@ -1,7 +1,24 @@
-FROM node:current
+FROM node:12
 WORKDIR /app
 COPY package.json /app
-RUN npm install
+COPY package-lock.json /app
```

Comment:
That's a good idea, and there are probably several RUN steps that could be combined into a single bash script to avoid creating more layers than are likely to be useful.